notebooks/logging-examples/logging-experiment-failures.ipynb

```diff
@@ -0,0 +1,157 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "3b466d92",
+   "metadata": {},
+   "source": [
+    "## Distinguishing Failed Experiments with Rubicon \n",
+    "Modeling runs can fail for a number of reasons. When logging with rubicon_ml, a failed run may result in an empty or incomplete experiment. In this example, we'll walk through how to handle such experiments."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d4f58990",
+   "metadata": {},
+   "source": [
+    "First lets simulate the problem. To do this we'll create an estimator that will fail on it's `fit()` 30% of the time.  We will consider any experiment that has a `none` value for `self.state`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "f9a506c3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.base import BaseEstimator\n",
+    "import random\n",
+    "\n",
+    "class BadEstimator(BaseEstimator):\n",
+    "    def __init__(self, state=None):\n",
+    "        super().__init__()\n",
+    "        self.state=None\n",
+    "        self.knn = KNeighborsClassifier(n_neighbors=2)\n",
+    "    def fit(self, X, y):\n",
+    "        self.knn.fit(X, y)\n",
+    "        output=random.random()\n",
+    "        if output<.3:\n",
+    "            output=None\n",
+    "        self.state=output\n",
```

Comment:
`state` is a learned attribute, so it should be `self.state_` per sklearn standards

also we should just say 
```
if output > .3
    self.state_ = output
````

this way `self.state_` is the random value if its greater than 0.3 and thus a "success," or it just doesn't exist if the random value is less than 0.3 and thus a "failure"

Reply:
Since we're looping and fitting multiple times, I'm keeping resetting it to `none` otherwise it would keep the `state
_` value of the last success